src/persist-client/src/internal/state_diff.rs

```diff
@@ -397,7 +394,8 @@ impl<T: Timestamp + Lattice + Codec64> State<T> {
                     // issues that may arise from diff application. We pass along the original
                     // Bytes it decoded from just so we can decode in this error path, while
                     // avoiding any extraneous clones in the expected Ok path.
-                    let diff = StateDiff::<T>::decode(&self.applier_version, data);
+                    // FIXME: this passes the state version but the method requires the build version.
+                    let diff = StateDiff::<T>::decode(&self.collections.version, data);
```

Comment:
Why can't we pass the applier_version here? That is set from the build version, no? In many cases they will be equal but will diverge during upgrades.